packages/components/src/components/Menu/Menu.tsx

```diff
@@ -168,275 +41,142 @@ const MenuItem = ({
-        );
-
-        const visibleItems = items?.map(group => ({
-            ...group,
-            options: group.options.filter(item => !item.isHidden),
-        }));
-
-        const flatGroupItems = flattenVisibleItems(visibleItems, addon);
-
-        // handle selecting an item
-        useEffect(() => {
-            const handleKeyDown = (e: KeyboardEvent) => {
-                if (!flatGroupItems || !flatGroupItems.length || focusedItemIndex === null) {
-                    return;
-                }
-
-                if (e.key === ' ' || e.key === 'Enter') {
-                    e.preventDefault();
-
-                    const focusedItem = flatGroupItems[focusedItemIndex];
-                    if (focusedItem.id === 'addon') {
-                        addon?.onClick?.();
-                    } else {
-                        focusedItem?.onClick?.();
-                    }
-
-                    if (focusedItem.shouldCloseOnClick !== false) {
-                        setToggled(false);
-                    }
-                }
-            };
-
-            if (focusedItemIndex !== null && flatGroupItems?.length) {
-                document.addEventListener('keydown', handleKeyDown);
+export const Menu = forwardRef<HTMLUListElement, MenuProps>(({ items, content, onClose }, ref) => {
+    const visibleItems = items?.filter(item => !item.isHidden);
+    const [focusedItemIndex, setFocusedItemIndex] = useState(
+        visibleItems?.length ? visibleItems.findIndex(item => !item.isDisabled) : null,
+    );
 
-                return () => {
-                    document.removeEventListener('keydown', handleKeyDown);
-                };
+    // handle selecting an item
+    useEffect(() => {
+        const handleKeyDown = (e: KeyboardEvent) => {
+            if (!visibleItems || !visibleItems.length || focusedItemIndex === null) {
+                return;
             }
-        }, [focusedItemIndex, flatGroupItems, setToggled, addon]);
 
-        // handle keyboard navigation
-        useEffect(() => {
-            const handleKeyDown = (e: KeyboardEvent) => {
-                if (
-                    (e.key === 'ArrowUp' || e.key === 'ArrowDown') &&
-                    flatGroupItems &&
-                    flatGroupItems.length > 0 &&
-                    focusedItemIndex !== null
-                ) {
-                    e.preventDefault();
-                    setFocusedItemIndex(getNextIndex(e.key, flatGroupItems));
-                }
-            };
+            if (e.key === ' ' || e.key === 'Enter') {
+                e.preventDefault();
 
-            if (focusedItemIndex !== null && flatGroupItems?.length) {
-                document.addEventListener('keydown', handleKeyDown);
-
-                return () => {
-                    document.removeEventListener('keydown', handleKeyDown);
-                };
+                const focusedItem = visibleItems[focusedItemIndex];
+                focusedItem?.onClick?.();
```

Comment:
I think this is kinda werid behavior, I tried it now, and:
- if I click the item (I tried the export menu), it submits the action and closes the dropdown
- if I use arrows up and down to select it and hit enter to submit the action, the dropdown isn't closed